config/config_test.go

```diff
@@ -1 +1,476 @@
 package config
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"reflect"
+	"testing"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/client/unversioned/testclient"
+	"k8s.io/kubernetes/pkg/util/intstr"
+)
+
+func testDir(t *testing.T) (string, func()) {
+	dir, err := ioutil.TempDir("", "haproxy_config")
+	if err != nil {
+		t.Fatal("failed to build dir")
+	}
+	return dir, func() { defer os.RemoveAll(dir) }
```

Comment:
Shouldn't this be just `func() { os.RemoveAll(dir) }` if you're going to call it in a defer below ?